### PR TITLE
Bug 1216163 - Add 'disabling' state. r=michielbdejong, yifan

### PR DESCRIPTION
**Pull request by ferjm**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d33a6286ce035c1921e1e6f5d811d0f3424b2168

File: apps/system/test/unit/sync_state_machine_test.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ suite('system/SyncStateMachine >', () => {
   suite('Events invalid for state', () => {
```
Comment (michielbdejong) [inline apps/system/test/unit/sync_state_machine_test.js]: Nit: Events -> Transitions

Transitions are:
* enable
* success
* sync
* disable
* error

Events (states) are:
* disabled
* enabling
* enabled
* syncing
* disabling
* errored

File: apps/system/test/unit/sync_state_machine_test.js [hunk 2]
```diff
@@ -64,35 +64,43 @@ suite('system/SyncStateMachine >', () => {
...
+      invalidTransitions: [
+        'enable',
+        'disable',
+        'sync'
+      ],
+      transition: 'success'
     }].forEach(config => {
-      test(config.from + ' - invalid events', (done) => {
-        config.invalidEvents.forEach(event => {
+      test(config.from + ' - invalid events', done => {
```
Comment (michielbdejong) [inline apps/system/test/unit/sync_state_machine_test.js]: Nit: events -> transitions

File: apps/system/test/unit/sync_state_machine_test.js [hunk 3]
```diff
@@ -64,35 +64,43 @@ suite('system/SyncStateMachine >', () => {
...
+        config.invalidTransitions.forEach(transition => {
           try {
-            Service.request('SyncStateMachine:'+ event);
+            Service.request('SyncStateMachine:'+ transition);
             assert.ok(false, 'Should have thrown exception');
           } catch(e) {
             assert.ok(true, 'Expected exception');
-            assert.equal(e.message, 'Event ' + event +
+            assert.equal(e.message, 'Event ' + transition +
```
Comment (michielbdejong) [inline apps/system/test/unit/sync_state_machine_test.js]: Nit: Event -> Transition

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=db783e289978ddb68d4e894a218eacf5ed8ab7eb

**Pull request merged**